sql/pgwire/v3.go

```diff
@@ -622,8 +622,8 @@ func (c *v3Conn) sendResponse(resp sql.Response, formatCodes []formatCode, sendD
 		return c.sendCommandComplete(nil)
 	}
 	for _, result := range resp.Results {
-		if result.PErr != nil {
-			if err := c.sendError(result.PErr.String()); err != nil {
+		if result.Err != nil {
+			if err := c.sendError(result.Err.Error()); err != nil {
```

Comment:
Hmm, I see. Don't have a strong opinion as I haven't yet even finished the conversion between `error` and `roachpb.Error`, but it might become more difficult to reason if we have one more error type. On the other hand, I couldn't convert `roachpb.Error` to `error` in several places when methods need to call `getDetail` for simple type check. A new simplified error might help. 

Meanwhile, I'll close this PR and revert #4479.